modules/nf-core/sratools/fasterqdump/main.nf

```diff
@@ -24,7 +24,17 @@ process SRATOOLS_FASTERQDUMP {
     def args2 = task.ext.args2 ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
     def outfile = meta.single_end ? "${prefix}.fastq" : prefix
-    def key_file = certificate ? "--perm ${certificate}" : ''
+    def key_file = ''
+
+    if (certificate){
+        if (certificate.toString().endsWith('.jwt')){
+            key_file += " --perm ${certificate}"
+            }
+        else if (certificate.toString().endsWith('.ngc')){
+            key_file += " --ngc ${certificate}"
+        }
+    }
```

Comment:
You could reorganize this, but it's also fine to leave it as you have done.

```suggestion
        if (certificate.extension == 'jwt') {
            key_file += " --perm ${certificate}"
        }
        else if (certificate.extension == 'ngc') {
            key_file += " --ngc ${certificate}"
        }
```

Reply:
Thanks for the comment. Removed top level if statement in latest commit.